Automattic/jetpack
# jetpack: Work around another calypso-build oddity

#### Changes proposed in this Pull Request:

Calypso-build sets webpack's `output.libraryTarget` to "window". When no
library name is supplied (which calypso-build doesn't do), this means
that everything exported by the module will be copied onto the `window`
object.

This could be bad, if we were exporting a bunch of stuff. But most of
our webpack entry points don't actually export anything, they're empty
ES modules.

Under Webpack 4, this worked out ok. All zero of the properties of the
empty ES modules got copied to `window`, so nothing really happened.

But Webpack 5 changed the behavior slightly: in addition to copying all
zero of the exports, it also flags the target object (`window`) as being
an ES module. This breaks other ES modules that do things like `import
window from 'global/window'`, as that'll start treating `window` as an
ES module (even though it lacks the correct structure) instead of as a
legacy object, leading to the breakage described in Automattic/wp-calypso#55696.

The fix is simple enough: Override calypso-build to set the library
target to the empty string, so it doesn't try to copy the zero
properties anywhere. The few entry points we have that were relying
on this behavior were all only exporting a single thing, so we can
explicitly set `library.name` and `library.export` to get the desired
behavior.

I also deleted one export that seemed clearly erroneous, since it's
unlikely it really intended to create `window.initialize()` and
internally it assigned the same function to
`global.jetpackSearchConfigureInit`.

#### Jetpack product discussion


https://github.com/Automattic/wp-calypso/issues/55696#issuecomment-915430357

#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:






* Try reproducing according to instructions at Automattic/wp-calypso#55696.
  * In my case, I managed to reproduce by [disabling the widget block editor](https://developer.wordpress.org/block-editor/how-to-guides/widgets/opting-out/#using-a-filter) then adding a Video widget, and also having Instant Search enabled.
* Make sure the blocks themselves and so on still work. Particularly you might want to look at the Instant Search customizer, the admin page, and the Widget Visibility thing, as those were the ones that had exports previously.
